robyn/__init__.py

```diff
@@ -111,86 +103,64 @@ def start(self, url: str = "127.0.0.1", port: int = 8080):
-                        copied_socket,
-                        workers,
-                    ),
-                )
-                process.start()
-                process_pool.append(process)
-
-            return process_pool
+        logger.info(f"Starting server at {url}:{port}")
 
         mp.allow_connection_pickling()
 
-        if not self.dev:
-            workers = self.workers
-            socket = SocketHeld(url, port)
-
-            process_pool = init_processpool(socket)
+        if not self.config.dev:
+            run_processes(
+                url,
+                port,
+                self.directories,
+                self.request_headers,
+                self.router.get_routes(),
+                self.middleware_router.get_routes(),
+                self.web_socket_router.get_routes(),
+                self.event_handlers,
+                self.config.workers,
+                self.config.processes,
+            )
+        else:
+            event_handler = EventHandler(
+                url,
+                port,
+                self.directories,
+                self.request_headers,
+                self.router.get_routes(),
+                self.middleware_router.get_routes(),
+                self.web_socket_router.get_routes(),
+                self.event_handlers,
+                self.config.workers,
+                self.config.processes,
+            )
+            event_handler.start_server()
+            logger.info(
+                f"Dev server initialized with the directory_path : {self.directory_path}",
+                Colors.BLUE,
+            )
 
             def terminating_signal_handler(_sig, _frame):
-                logger.info(
-                    f"{Colors.BOLD}{Colors.OKGREEN} Terminating server!! {Colors.ENDC}"
-                )
-                for process in process_pool:
-                    process.kill()
+                logger.info("Terminating server!!", bold=True)
+                event_handler.stop_server()
+                observer.stop()
+                observer.join()
 
             signal.signal(signal.SIGINT, terminating_signal_handler)
             signal.signal(signal.SIGTERM, terminating_signal_handler)
 
-            logger.info(f"{Colors.OKGREEN}Press Ctrl + C to stop \n{Colors.ENDC}")
-            for process in process_pool:
-                process.join()
-        else:
-            event_handler = EventHandler(self.file_path)
-            event_handler.start_server_first_time()
-            logger.info(
-                f"{Colors.OKBLUE}Dev server initialised with the directory_path : {self.directory_path}{Colors.ENDC}"
-            )
             observer = Observer()
             observer.schedule(event_handler, path=self.directory_path, recursive=True)
             observer.start()
+
             try:
-                while True:
-                    pass
+                while observer.is_alive():
+                    observer.join(1)
```

Comment:
@AntoineRR , what does this do?

Reply:
I took it from the quickstart example in the doc -> https://python-watchdog.readthedocs.io/en/stable/quickstart.html#quickstart
My PC fans are doing way better since I added it, as the CPU is not constantly at 100% with this :smile:

Reply:
Aha, perfect 😄